controller/setting_controller.go

```diff
@@ -1062,3 +1071,151 @@ func (sc *SettingController) cleanupFailedSupportBundles() error {
 
 	return nil
 }
+
+func (sc *SettingController) GetCheckUpgradeRequestExtraInfo() (CheckUpgradeRequestExtraInfo, error) {
+	extraInfo := &ExtraInfo{
+		logger:        sc.logger,
+		ds:            sc.ds,
+		kubeClient:    sc.kubeClient,
+		metricsClient: sc.metricsClient,
+		structFields:  util.StructFields{},
+		controllerID:  sc.controllerID,
+		namespace:     sc.namespace,
+	}
+
+	kubeVersion, err := sc.kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Kubernetes server version")
+	}
+	extraInfo.structFields.Append(ExtraInfoKubernetesVersion, kubeVersion.GitVersion, extraInfo.structFields.ConvertTag(ExtraInfoKubernetesVersion))
+
+	extraInfo.collectNodeScope()
+
+	responsibleNodeID, err := getResponsibleNodeID(sc.ds)
+	if err != nil {
+		sc.logger.WithError(err).Warn("Failed to select node to get extra info")
+	}
+	if responsibleNodeID == sc.controllerID {
+		extraInfo.collectClusterScope()
+	}
+
+	return extraInfo.structFields.NewStruct(), nil
+}
+
+const (
+	ExtraInfoKubernetesVersion      = util.StructName("KubernetesVersion")
+	ExtraInfoKubernetesNodeProvider = util.StructName("KubernetesNodeProvider")
+
+	ExtraInfoHostKernelRelease = util.StructName("HostKernelRelease")
+	ExtraInfoHostOsDistro      = util.StructName("HostOsDistro")
+
+	ExtraInfoNamespaceUID = util.StructName("LonghornNamespaceUid")
+	ExtraInfoNodeCount    = util.StructName("LonghornNodeCount")
```

Comment:
Can we have a comment to differentiate which are cluster or node level info? I know you have collectNode and collectCluster, but it would be helpful to organize the const as well.


```go
// Cluster info 
...
// Node info
...
```

For example, longhorn node count is cluster level info.